### PR TITLE
add azure integration

### PR DESCRIPTION
**Pull request by TimoKorinth**

### CODE REVIEW

File: configs/system/config.example.yaml [hunk 1]
```diff
@@ -20,7 +20,22 @@ features:
   # ─────────────────────── TTS Provider ─────────────────────────
   # You can override the text-to-spech provider if your Wingman supports it. Our OpenAI wingman does!
   # Note that the other providers may have additional config blocks as shown below for edge_tts. These are only used if the provider is set here.
-  tts_provider: openai # available: openai, edge_tts, elevenlabs
+  tts_provider: openai # available: openai, edge_tts, elevenlabs, azure
+
+  # ─────────────────────── Speech to text Provider ─────────────────────────
```
Comment (Shackless) [inline configs/system/config.example.yaml:25]: No copy&paste docs here please. Explain briefly what these different settings are/do.

File: configs/system/config.example.yaml [hunk 2]
```diff
@@ -111,6 +126,28 @@ elevenlabs:
...
     use_speaker_boost: true # true or false
 
+# ────────────────────────────────── AZURE ────────────────────────────────────
+# Azure is a paid subscription provider from Microsoft which also offers OpenAI API access.
+# If you configured some providers above to use Azure, you need to provide your Azure settings here.
+# Please also provide your Azure API keys in the secrets.yaml.
+azure:
+  whisper:
+    api_base_url: xxx # https://xxx.openai.azure.com/
+    api_version: 2023-09-01-preview # The API version to use. For a list of supported versions, see here: https://learn.microsoft.com/en-us/azure/ai-services/openai/reference
```
Comment (Shackless) [inline configs/system/config.example.yaml:136]: same here, move common docs up before the `azure` block

**Review by Shackless — APPROVED** [review]

**Comment by bukit-kronik** [comment]
Just curious, will it support custom neural voice?

**Comment by TimoKorinth** [comment]
I haven't tested this feature, because we have to apply for this feature separately, but as the python lib supports custom voices, it should work, yes.

**Pull request merged**